framework/src/fviks/FVInterfaceKernel.C

```diff
@@ -0,0 +1,177 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "FVInterfaceKernel.h"
+#include "FEProblemBase.h"
+#include "SystemBase.h"
+#include "MooseVariableFV.h"
+#include "Assembly.h"
+
+InputParameters
+FVInterfaceKernel::validParams()
+{
+  InputParameters params = MooseObject::validParams();
+  params += TransientInterface::validParams();
+  params += BoundaryRestrictableRequired::validParams();
+  params += TaggingInterface::validParams();
+  params += TwoMaterialPropertyInterface::validParams();
+
+  params.addRequiredParam<std::vector<SubdomainName>>(
+      "subdomain1", "The subdomains on the 1st side of the boundary.");
+  params.addRequiredParam<std::vector<SubdomainName>>(
+      "subdomain2", "The subdomains on the 2nd side of the boundary.");
+  params.addRequiredParam<NonlinearVariableName>(
+      "variable1", "The name of the first variable that this interface kernel applies to");
+  params.addParam<NonlinearVariableName>(
+      "variable2",
+      "The name of the second variable that this interface kernel applies to. If not supplied, "
+      "variable1 will be used.");
+  params.addParam<bool>("use_displaced_mesh",
+                        false,
+                        "Whether or not this object should use the "
+                        "displaced mesh for computation.  Note that "
+                        "in the case this is true but no "
+                        "displacements are provided in the Mesh block "
+                        "the undisplaced mesh will still be used.");
+
+  params.addParamNamesToGroup("use_displaced_mesh", "Advanced");
+  params.addCoupledVar("displacements", "The displacements");
+  params.declareControllable("enable");
+  params.registerBase("FVInterfaceKernel");
+  params.registerSystemAttributeName("FVInterfaceKernel");
+  return params;
+}
+
+FVInterfaceKernel::FVInterfaceKernel(const InputParameters & parameters)
+  : MooseObject(parameters),
+    BoundaryRestrictableRequired(this, false),
+    SetupInterface(this),
+    FunctionInterface(this),
+    DistributionInterface(this),
+    UserObjectInterface(this),
+    TransientInterface(this),
+    PostprocessorInterface(this),
+    VectorPostprocessorInterface(this),
+    GeometricSearchInterface(this),
+    Restartable(this, "FVInterfaceKernels"),
```

Comment:
Do interface kernels need this?
I would think that variables may be restartable objects, but the kernels don't hold things we d want to restart

Same question for FVBCs I guess

Reply:
A great question. We always want to make sure that our residuals aren't stateful (e.g. we should always get the same residual for the same supplied solution vector) and having restartable (potentially stateful) probably shouldn't be encouraged. I'll remove this